lms/djangoapps/discussion_api/api.py

```diff
@@ -556,6 +552,12 @@ def _handle_voted_field(form_value, cc_content, api_content, request, context):
     )
 
 
+def _handle_read_field(form_value, user, cc_content):
+    """mark thread as read for the user"""
+    if form_value:
+        user.read(cc_content)
```

Comment:
I don't think it is a feature we need right now, but @scottrish can speak
to that.

On Wed, Jun 1, 2016 at 4:10 AM, wajeeha-khalid <notifications@github.com>
wrote:

> In lms/djangoapps/discussion_api/api.py
> <https://github.com/edx/edx-platform/pull/12568#discussion_r65318382>:
>
> > @@ -556,6 +552,12 @@ def _handle_voted_field(form_value, cc_content, api_content, request, context):
> >      )
> >
> >
> > +def _handle_read_field(form_value, user, cc_content):
> > +    """mark thread as read for the user"""
> > +    if form_value:
> > +        user.read(cc_content)
>
> @jcdyer <https://github.com/jcdyer> Thread is only marked as unread when
> anyone add response/comment, report/Unreport, vote/unvote and thats being
> handled implicitly. There is no explicit UI to mark thread as unread (as
> far as current behaviour is concerned).
>
> May be we can add a business feature to allow user to explicitly mark
> thread back to unread. What do you think?
> cc: @robrap <https://github.com/robrap> @scottrish
> <https://github.com/scottrish>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/edx/edx-platform/pull/12568/files/4abc621c7e139aceb61a91449b6e82e8a8f4440e..52f91abfff35307ecca17da916d734f391e94d91#r65318382>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AN5rPdq9K_oOz7HKfFZQYtfhdPwTADX6ks5qHT6DgaJpZM4InbSL>
> .
>



-- 

*Robert Raposa*

edX | Software Architect  | rraposa@edx.org

141 Portland Street, 9th floor

Cambridge, MA 02139
http://www.edx.org <http://www.edxonline.org/>

[image:
http://www.e-learn.nl/media/blogs/e-learn/edX_Logo_Col_RGB_FINAL.jpg?mtime=1336074566]